File: CS3305/Demos/demo0/Counter.py
#! /usr/bin/env python3

class Counter:

    def __init__(self, counter = 0):
        self.theCounter = counter
    #end constructor

    def incr( self ):
        self.theCounter += 1
    #end incr

    def getValue( self ):
        return self.theCounter
    #end getValue

    def __str__( self ):
        return str( self.theCounter )
    #end __str__

# end class Counter

def main():
    print("%s" % "Testing the Counter class")

    counter = Counter()
    # implicit call to counter's __str__ method
    print( "%s %s" % ("Value of the counter after constructor ", counter) )

    for i in range(1,21): counter.incr()
    print( "%s %s" % ("Value of counter after calling incr 20 times ",
            counter) )
    print( "%s %d" % ("And value of counter from calling getValue ",
            counter.getValue() ) )

    print("%s" % "Let's add 50 more to the counter" )

    for i in range(1,51): counter.incr()

    print("%s %s" % ( "Value of counter after calling incr 50 more times ",
             counter ) )
    print( "%s %d" % ("And value of counter from calling getValue ",
             counter.getValue() ) )

    counter = Counter(100) #create new counter object with initial value 100
    print( "%s %s" % ("Value of the counter after constructor Counter(100) ", counter) )

# end main

if __name__ == "__main__":
    main()

File: CS3305/Demos/demo0/Counter.java

public class Counter {

    private int theCounter;

    public Counter() {
        this.theCounter = 0;
    }

    public Counter( int initValue ) {
        this.theCounter = initValue;
    }

    public void incr( ) {
        this.theCounter++;
    }

    public int getValue( ) {
        return this.theCounter;
    }

    public String toString( ) {
        StringBuffer sb = new StringBuffer();
        sb.append( this.theCounter );
        return sb.toString();
    }

    public static void main( String[] args ) {
        System.out.printf("%s\n", "Testing the Counter class ");

        Counter counter = new Counter( );
        System.out.printf("%s %s\n", "Value of the counter after constructor ",
                             counter );  //implicit call to counter.toString()

        for (int i=1; i <= 20; i++) counter.incr( );
        System.out.printf( "%s %s\n", "Value of counter after calling incr 20 times ",
                            counter ); // implicit call to counter.toString()
        System.out.printf("%s %d\n", "And value of counter from calling getValue ", 
                            counter.getValue( ) ); 

        System.out.printf("%s\n",  "Let's add 50 more to the counter" );
        for (int i=1; i <= 50; i++) counter.incr( );
        System.out.printf("%s %s\n",  "Value of counter after calling incr 50 more times ",
                            counter ); // implicit call to counter.toString()
        System.out.printf("%s %d\n", "And value of counter from calling getValue ",
                            counter.getValue( ) ); 

        counter = new Counter(100);
        System.out.printf("%s %s\n", "Value of the counter after constructor new Counter(100)",
                             counter );  //implicit call to counter.toString()

    }

}
